game/controllers-midi.cc

```diff
@@ -3,22 +3,22 @@
 #include "controllers.hh"
 #include "portmidi.hh"
 #include "fs.hh"
-#include <boost/regex.hpp>
+#include <regex>
 #include <unordered_map>
 
 namespace input {
 
 	class Midi: public Hardware {
 	public:
 		Midi() {
-			boost::regex re(config["game/midi_input"].s());
+			std::regex re(config["game/midi_input"].s());
 			for (int dev = 0; dev < Pm_CountDevices(); ++dev) {
 				try {
 					PmDeviceInfo const* info = Pm_GetDeviceInfo(dev);
 					if (!info->input) continue;  // Not an input device
 					if (info->opened) continue;  // Already opened
 					std::string name = getName(dev);
-					if (!re.empty() && !regex_search(name, re)) continue;
```

Comment:
isn't it a little dangerous to skip the empty check here?

Reply:
Like i said on discord. The std:basic_regex has no empty method anymore see this paper: http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2003/n1507.htm

If a std::basic_regex (std::regex is derived from it) is filled with an empty string it will not match anything. So it wouldn't matter much

Reply:
ok I wasn't able to read the discord due to MIA last weekend.